apps/meteor/client/views/navigation/providers/RoomsNavigationProvider.tsx

```diff
@@ -0,0 +1,271 @@
+import { useQueuedInquiries } from '../../../hooks/omnichannel/useQueuedInquiries';
+import { useSortQueryOptions } from '../../../hooks/useSortQueryOptions';
+import { RoomsNavigationContext, SIDE_PANEL_GROUPS, SIDE_BAR_GROUPS } from '../contexts/RoomsNavigationContext';
+import { useCollapsedGroups } from '../hooks/useCollapsedGroups';
+import { useSidePanelFilters } from '../hooks/useSidePanelFilters';
+
+const query = { open: { $ne: false } };
+
+const emptyQueue: ILivechatInquiryRecord[] = [];
+
+export type GroupedUnreadInfoData = {
+	userMentions: number;
+	groupMentions: number;
+	tunread: string[];
+	tunreadUser: string[];
+	unread: number;
+};
+
+export type useRoomsGroupsReturnType = {
+	sideBar: {
+		roomList: Array<SubscriptionWithRoom>;
+		groupsCount: number[];
+		groupsList: TranslationKey[];
+		groupedUnreadInfo: GroupedUnreadInfoData[];
+	};
+	sidePanel: {
+		roomList: { [key: string]: SubscriptionWithRoom[] };
+		groupedUnreadInfo: { [key: string]: GroupedUnreadInfoData };
+	};
+};
+
+const getUnreadInfo = (groupData: Set<any>, showUnreadInfo = true) => {
+	const groupedUnreadInfoAcc = {
+		userMentions: 0,
+		groupMentions: 0,
+		tunread: [],
+		tunreadUser: [],
+		unread: 0,
+	};
+
+	if (!showUnreadInfo) {
+		return groupedUnreadInfoAcc;
+	}
+
+	const groupedUnreadInfo = [...groupData].reduce(
+		(counter, { userMentions, groupMentions, tunread, tunreadUser, unread, alert, hideUnreadStatus }) => {
+			if (hideUnreadStatus) {
+				return counter;
+			}
+
+			counter.userMentions += userMentions || 0;
+			counter.groupMentions += groupMentions || 0;
+			counter.tunread = [...counter.tunread, ...(tunread || [])];
+			counter.tunreadUser = [...counter.tunreadUser, ...(tunreadUser || [])];
+			counter.unread += unread || 0;
+			!unread && !tunread?.length && alert && (counter.unread += 1);
+			return counter;
+		},
+		groupedUnreadInfoAcc,
+	);
+
+	return groupedUnreadInfo;
+};
+
+const isUnread = (room: SubscriptionWithRoom) => (room.alert || room.unread || room.tunread?.length) && !room.hideUnreadStatus;
+
+const hasMention = (room: SubscriptionWithRoom) => room.userMentions || room.groupMentions || room.tunreadUser || room.tunreadGroup;
+
+export const useSidePanelGroups = ({ onlyUnReads }: { onlyUnReads?: boolean }) => {
+	const groups = useRoomsGroups();
+
+	const { roomList, groupedUnreadInfo } = useMemo(
+		() =>
+			Object.values(SIDE_PANEL_GROUPS).reduce(
+				(acc, groupTitle) => {
+					const value = groups.get(groupTitle) || new Set();
+
+					const roomList = onlyUnReads ? [...value].filter((room) => isUnread(room)) : [...value];
+					const groupedUnreadInfo = getUnreadInfo(value);
+
+					acc.groupedUnreadInfo[groupTitle] = groupedUnreadInfo;
+					acc.roomList[groupTitle] = roomList;
+					return acc;
+				},
+				{
+					roomList: {},
+					groupedUnreadInfo: {},
+				} as useRoomsGroupsReturnType['sidePanel'],
+			),
+		[groups, onlyUnReads],
+	);
+
+	return { roomList, groupedUnreadInfo };
+};
```

Comment:
I think we can leverage the group sets and maps to avoid this computation. Let's discuss it better in a call.

Reply:
fixed!